com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/parser/IdleUriParametersParser.java

```diff
@@ -1,16 +1,74 @@
 package com.sap.cloud.lm.sl.cf.core.parser;
 
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
 import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+import com.sap.cloud.lm.sl.cf.core.util.ApplicationURI;
+import com.sap.cloud.lm.sl.cf.core.util.UriUtil;
+import com.sap.cloud.lm.sl.cf.core.validators.parameters.RoutesValidator;
+import com.sap.cloud.lm.sl.mta.util.PropertiesUtil;
 
 public class IdleUriParametersParser extends UriParametersParser {
 
     public IdleUriParametersParser(String defaultHost, String defaultDomain, String routePath) {
-        super(defaultHost, defaultDomain, SupportedParameters.IDLE_HOST, SupportedParameters.IDLE_DOMAIN, true, routePath);
+        super(defaultHost, defaultDomain, SupportedParameters.IDLE_HOST, SupportedParameters.IDLE_DOMAIN, routePath);
     }
 
     public IdleUriParametersParser(String defaultHost, String defaultDomain, String hostParameterName, String domainParameterName,
                                    String routePath) {
-        super(defaultHost, defaultDomain, hostParameterName, domainParameterName, true, routePath);
+        super(defaultHost, defaultDomain, hostParameterName, domainParameterName, routePath);
+    }
+
+    @Override
+    public List<String> getApplicationRoutes(List<Map<String, Object>> parametersList) {
+        List<String> idleRoutes = getIdleRoutes(parametersList);
+        if (!idleRoutes.isEmpty()) {
+            return idleRoutes;
+        }
+
+        List<String> liveRoutes = super.getApplicationRoutes(parametersList);
+        if (!liveRoutes.isEmpty()) {
+            return modifyLiveRoutes(liveRoutes);
+        }
+        return Collections.emptyList();
+    }
+
+    private List<String> getIdleRoutes(List<Map<String, Object>> parametersList) {
+        List<Map<String, Object>> idleRoutesMaps = RoutesValidator.applyRoutesType(PropertiesUtil.getPropertyValue(parametersList,
+                                                                                                                   SupportedParameters.IDLE_ROUTES,
+                                                                                                                   null));
+        return idleRoutesMaps.stream()
+                             .map(routesMap -> (String) routesMap.get(SupportedParameters.IDLE_ROUTE))
+                             .filter(Objects::nonNull)
+                             .map(UriUtil::stripScheme)
```

Comment:
There are not unit test about these two cases. 
This is because IdleUriParametersParserTest constructs properly idle-routes parameter.
Not sure if this is tested somewhere else?